src/Relude/Extra/Foldable1.hs

```diff
@@ -107,6 +107,24 @@ class Foldable f => Foldable1 f where
     minimum1 :: Ord a => f a -> a
     minimum1 = SG.getMin #. foldMap1 SG.Min
 
+    {- | The largest element of a non-empty data structure
+         with respect to the given comparison function.
+
+    >>> maximumOn1 sin (32 :| [64, 8, 128, 16])
+    8.0
+    -}
+    maximumOn1 :: Ord b => (a -> b) -> f a -> a
+    maximumOn1 f = maximumOn1 f . toNonEmpty
+
+    {- | The smallest element of a non-empty data structure
+         with respect to the given comparison function.
+
+    >>> minimumOn1 sin (32 :| [64, 8, 128, 16])
+    16.0
+    -}
+    minimumOn1 :: Ord b => (a -> b) -> f a -> a
+    minimumOn1 f = minimumOn1 f . toNonEmpty
```

Comment:
Let's add `{-# INLINE #-}` pragmas to default implementations of `maximumOn1` and `minimumOn1`